base/src/main/scala/data/MaybeFunctions.scala

```diff
@@ -1,16 +1,18 @@
 package scalaz
 package data
 
+import typeclass.Liskov._
+
 import Maybe.{Empty, Just}
 
 trait MaybeFunctions {
-  def empty[A]: Maybe[A] = Empty()
+  def empty[A]: Maybe[A] = liftCvf[Nothing, A, Maybe](isa[Nothing, A]).apply(Empty)
   def just[A](a: A): Maybe[A] = Just(a)
 
   def maybe[A, B](n: B)(f: A => B): Maybe[A] => B = _ match {
-    case Empty() => n
-    case Just(x) => f(x)
+    case Empty    => n
```

Comment:
NB: A ~ Nothing in this context. As long as you're comfortable with that, liftCvf is fine.